### DISCUSSION TITLE
Can't get values from IngestJobV2SuccessfulResults.

### DISCUSSION THREAD

**Discussion by watanabeshuji** [jsforce/jsforce#1635] [Q&A]
I use bulk v2 API and get result from `IngestJobV2SuccessfulResults`.
But I can't get ExternalId__c value from that result.

I think these code is strange.
https://github.com/jsforce/jsforce/blob/main/src/api/bulk2.ts#L69-L74

I want to access such as this code.
```
for (const result of successfulResults) {
   const extId = result[externalIdFieldName];
}
```

Is it correct?
```
export type IngestJobV2SuccessfulResults = Array<
  {
    sf__Created: 'true' | 'false';
    sf__Id: string;
    [name: string]: string;
  }
>;
```

**Comment by cristiand391** [comment]
Hey, sorry for the late reply
The `IngestJobV2SuccessfulResults` type is correct, the `S` refers to the record type specified when creating the `Connection` instance:

https://github.com/jsforce/jsforce/blob/80a0483846232be58afa5930c1c6e0b592526a31/src/api/bulk2.ts#L73

https://jsforce.github.io/jsforce/classes/connection.Connection.html

### conn object with no record type:
![Screenshot 2025-01-06 at 11 03 02 AM](https://github.com/user-attachments/assets/a244f133-24ac-49ff-b8ee-35fd93117879)

### conn with record type extending `Schema`:
![Screenshot 2025-01-06 at 11 04 47 AM](https://github.com/user-attachments/assets/d5388e36-e35e-4cca-aee8-88ecd802df80)
